homeassistant/components/rainmachine/__init__.py

```diff
@@ -215,194 +272,68 @@ async def unpause_watering(call):
-        await asyncio.gather(*tasks)
-
-    async def async_update_sensors(self):
-        """Update sensor/binary sensor data."""
-        _LOGGER.debug("Updating sensor data for RainMachine")
-
-        # Fetch an API category if there is at least one interested entity:
-        tasks = {}
-        for category, count in self._api_category_count.items():
-            if count == 0:
-                continue
-            tasks[category] = self.async_fetch_from_api(category)
-
-        results = await asyncio.gather(*tasks.values(), return_exceptions=True)
-        for api_category, result in zip(tasks, results):
-            if isinstance(result, RainMachineError):
-                _LOGGER.error(
-                    "There was an error while updating %s: %s", api_category, result
-                )
-                continue
-
-        async_dispatcher_send(self.hass, SENSOR_UPDATE_TOPIC)
-
-    async def async_update_programs_and_zones(self):
-        """Update program and zone data.
-
-        Program and zone updates always go together because of how linked they are:
-        programs affect zones and certain combinations of zones affect programs.
-
-        Note that this call does not take into account interested entities when making
-        the API calls; we make the reasonable assumption that switches will always be
-        enabled.
-        """
-        _LOGGER.debug("Updating program and zone data for RainMachine")
-
-        tasks = {
-            DATA_PROGRAMS: self.async_fetch_from_api(DATA_PROGRAMS),
-            DATA_ZONES: self.async_fetch_from_api(DATA_ZONES),
-            DATA_ZONES_DETAILS: self.async_fetch_from_api(DATA_ZONES_DETAILS),
-        }
-
-        results = await asyncio.gather(*tasks.values(), return_exceptions=True)
-        for api_category, result in zip(tasks, results):
-            if isinstance(result, RainMachineError):
-                _LOGGER.error(
-                    "There was an error while updating %s: %s", api_category, result
-                )
+    await hass.config_entries.async_reload(entry.entry_id)
 
-        async_dispatcher_send(self.hass, PROGRAM_UPDATE_TOPIC)
-        async_dispatcher_send(self.hass, ZONE_UPDATE_TOPIC)
 
-
-class RainMachineEntity(Entity):
+class RainMachineEntity(CoordinatorEntity):
     """Define a generic RainMachine entity."""
 
-    def __init__(self, rainmachine):
+    def __init__(
+        self, coordinator: DataUpdateCoordinator, controller: Controller
+    ) -> None:
         """Initialize."""
+        super().__init__(coordinator)
         self._attrs = {ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION}
+        self._controller = controller
         self._device_class = None
+        # The colons are removed from the device MAC simply because that value
+        # (unnecessarily) makes up the existing unique ID formula and we want to avoid
+        # a breaking change:
+        self._device_mac = controller.mac.replace(":", "")
```

Comment:
Let's rename this to `self._unique_id` to make it clear it should not be used as a mac address